spec/bundler/plugin/index_spec.rb

```diff
@@ -86,6 +86,12 @@
       expect(new_index.hook_plugins("after-bar")).to eq([plugin_name])
     end
 
+    it "only registers a gem once for an event" do
+      path = lib_path(plugin_name)
+      index.register_plugin(plugin_name, path.to_s, [path.join("lib").to_s], commands, sources, hooks)
```

Comment:
As a way of expressing the redundancy of the hooks listed, the `hooks` argument could be doubled:

```suggestion
      index.register_plugin(plugin_name,
                            path.to_s,
                            [path.join("lib").to_s],
                            commands,
                            sources,
                            hooks + hooks)
```